### ISSUE TITLE
Docsy with all forms of Google Analytics?

### ISSUE THREAD

**Issue by Symbolics** [google/docsy#487]
I just signed up for Google Analytics for our doc site, but don't seem to see any traffic. I ran a dead-link checker from a few locations, so there should be some measurable traffic showing up, but it's all still 0, even on the 'real-time' page on the Google Analytics site.

This isn't really an area I know a lot about, but I wonder if it might be because we're using "Google 4" (whatever that means). The id begins with "G-", and the example in the Docsy user guide has "UA-".

Anyone know if this could be it?

**Comment by LisaFC** [comment]
It *should* work with both the older and newer versions of Analytics (the Google 4 is the latest generation of Google Analytics) but I think the newer version only works with the synchronous Hugo template and we use the async.

Can you try adding your own version of this file to your project and replace
```
{{ template "_internal/google_analytics_async.html" . }}
```
with:

```
{{ template "_internal/google_analytics.html" . }}
```

https://github.com/google/docsy/blob/master/layouts/partials/head.html#L23

If that works I'll change it in the theme as the synchronous template should work with both old and new Analytics - will also update docs.

**Comment by Symbolics** [comment]
No joy; still no stats showing up.

**Comment by LisaFC** [comment]
So it can take up to 24 hours for data to show up in Google Analytics, maybe hang on until this evening and see if anything is appearing then? (also presumably your site actually has traffic....)

**Comment by LisaFC** [comment]
Though did you check the real-time reports page as well? I'll have a look and see if there's anything else that could be causing it - we're just using Hugo's analytics integration so it's not something specific to Docsy.

**Comment by Symbolics** [comment]
I did check the real-time dashboard. It should have traffic, as I've been running a few link checkers and generating my own manual traffic.

**Comment by LisaFC** [comment]
Aha - it seems like it's a bug in Hugo. I've found a workaround: https://gist.github.com/zjeaton/42246742cdaf2fb46400d04c2eba9a8a#file-analytics-gtag-html

**Comment by Symbolics** [comment]
Good catch. It seems that it's been fixed with gohugoio#8287, but that hasn't made it into a release yet. Perhaps with Hugo 0.82. I'll keep an eye on this and update here when it's working.

**Comment by Symbolics** [comment]
Confirmed fixed in Hugo 0.82.0. The partial needs to be updated as [described in the Hugo docs](https://gohugo.io/templates/internal#configure-google-analytics). This is the same change described earlier in this issue.
